sunpycube/cube/tests/test_cube_utils.py

```diff
@@ -115,13 +115,39 @@ def test_convert_slice():
 
 def test_pixelize_slice():
     sl = [((slice(1, None, 3), 3.2, slice(None, None, 2)), wt),
-          ((0.3 * u.deg, 0.0001 * u.mm, 5400 * u.arcsec), wm),
-          ((slice(None, 3 * u.deg, 2), slice(5 * u.nm, 6, 1 * u.nm),
-            slice(90 * u.arcmin, 300, 60)), wm)]
+          ((0.3 * u.deg, 2474.5 * u.deg, 10.75 * u.Angstrom), wm),
+          ((slice(None, 3 * u.deg, 2), slice(99.5 * u.deg, 124.5, 1 * u.deg),
+            slice(90 * u.Angstrom, 300, 60)), wm)]
 
     results = [(slice(1, None, 3), 3.2, slice(None, None, 2)),
                (0, 4950, 4),
-               (slice(None, 7, 5), slice(200, 250, 50), slice(4, 11, 2))]
+               (slice(None, 7, 5), slice(200, 250, 2), slice(400, 1450, 300))]
 
     for i in range(len(sl)):
         assert cu.pixelize_slice(*sl[i]) == results[i]
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    (cu._convert_cube_like_index_to_sequence_indices(5, np.array([8, 16, 24, 32])), (0, 5)),
+    (cu._convert_cube_like_index_to_sequence_indices(8, np.array([8, 16, 24, 32])), (1, 0)),
+    (cu._convert_cube_like_index_to_sequence_indices(20, np.array([8, 16, 24, 32])), (2, 4)),
+    (cu._convert_cube_like_index_to_sequence_indices(50, np.array([8, 16, 24, 32])), (3, 7)),
+])
+def test_convert_cube_like_index_to_sequence_indices(test_input, expected):
+    assert test_input == expected
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    (cu._convert_cube_like_slice_to_sequence_slices(slice(2, 5),
+                                                    np.array([8, 16, 24, 32])), (slice(0, 1), slice(2, 5))),
+    (cu._convert_cube_like_slice_to_sequence_slices(slice(5, 15), np.array(
+        [8, 16, 24, 32])), (slice(0, 2), [slice(5, 8), slice(0, 7)])),
+    (cu._convert_cube_like_slice_to_sequence_slices(slice(5, 16),
+                                                    np.array([8, 16, 24, 32])), (slice(0, 2), [slice(5, 8)])),
+    (cu._convert_cube_like_slice_to_sequence_slices(slice(5, 23), np.array(
+        [8, 16, 24, 32])), (slice(0, 3), [slice(5, 8), slice(0, 7)])),
+    (cu._convert_cube_like_slice_to_sequence_slices(slice(5, 100),
+                                                    np.array([8, 16, 24, 32])), (slice(0, 4), [slice(5, 8)])),
```

Comment:
No the tuple[0] is the sequence_slice which will be from (0, 4) and the tuple[1] is the list which consist of the slices of 1st and last cube which if it needs to be sliced.